### ISSUE TITLE
Inconsistent behavior on invalid config

### ISSUE THREAD

**Issue by arvigeus** [igorshubovych/markdownlint-cli#105]
.markdownlint.json

```json
{
  "extends": "this-file-does-not-exist.json"
}
```

Running `markdownlint --fix '**/*.md'` silently fails to load invalid file and goes to default config.

Running `markdownlint --fix '**/*.md' --config .markdownlint.json` displays an error message that `this-file-does-not-exist.json` is missing.

In my case I was trying to load a valid file that was missing, and it took me some time to realize that the problem is me, not `markdownlint` being lazy not loading `.markdownlint.json`. I think it should complain in this case, instead of straight up loading default config.

**Comment by victoriadrake** [comment]
Continuation of: https://github.com/igorshubovych/markdownlint-cli/commit/a1f9a15f1adea0e4c23e90d1145f8fca02bd8a76#r40919702

I also experienced `markdownlint` silently using a default config when I had it run as a pre-commit hook (pre-commit framework). I had passed in a non-existent file path to `--config` by mistake. Here’s what the hook looks like in `.pre-commit-config.yaml`:

```yaml
  - repo: local
    hooks:
      - id: markdownlint
        name: markdownlint
        description: "Lint Markdown files"
        entry: markdownlint '**/*.md' --fix --ignore node_modules --config "./.markdownlint.json"
        language: node
        types: [markdown]
```

In this scenario, `.markdownlint.json` did not exist in the project root. I had also tried it with a YAML formatted config file before realizing the erroneous file path.

Expected behavior: an error is raised when the config file does not exist, ie:

`Config file not found. Using default config...`
